homeassistant/components/thermostat/__init__.py

```diff
@@ -301,22 +306,32 @@ def turn_fan_off(self):
     @property
     def min_temp(self):
         """Return the minimum temperature."""
-        return convert(7, TEMP_CELSIUS, self.unit_of_measurement)
+        unit = self.unit_of_measurement
+        if not unit:
```

Comment:
You'd think, but the code was actually not setting units and expecting a
tuple (value, unit) back from the temperature converter before I touched
it. It then pulled the unit from the conversion.

On Aug 4, 2016 9:08 AM, "Paulus Schoutsen" <notifications@github.com> wrote:

> In homeassistant/components/thermostat/__init__.py
> <https://github.com/home-assistant/home-assistant/pull/2705#discussion_r73540451>
> :
>
> > @@ -301,22 +306,32 @@ def turn_fan_off(self):
> >      @property
> >      def min_temp(self):
> >          """Return the minimum temperature."""
> > -        return convert(7, TEMP_CELSIUS, self.unit_of_measurement)
> > +        unit = self.unit_of_measurement
> > +        if not unit:
>
> This should never happen. A thermostat *always* has a unit of measurement
> or else it would not be able to express any value.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/2705/files/dcac2dd629869be07e05bd8e4d7ac96956517873#r73540451>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AC2fZWRbxLgfTeVoLPzXhL37Pan2_IXRks5qcgBpgaJpZM4JaO1y>
> .
>